src/assets/source/scripts/modules/serlo_injections.js

```diff
@@ -80,21 +121,23 @@ define(['jquery', 'common', 'translator', 'content'], function ($, Common, t) {
                 };
 
                 // check if it is geogebra xml
-                if (data.documentElement && data.documentElement.nodeName === 'geogebra') {
-                    initGeogebraApplet(data.documentElement.outerHTML);
-                } else if (contentType === 'image/jpeg' || contentType === 'image/png') {
-                    $that.html('<img src="' + href + '" title="' + title + '" />');
-                } else {
-                    try {
-                        data = JSON.parse(data);
-                        if (data.response) {
-                            $that.html('<div class="panel panel-default"><div class="panel-body">' + data.response + '</div></div>');
-                            Common.trigger('new context', $that);
-                        } else {
+                if (! initGeogebraTube()) {
```

Comment:
I put the check for geogebratube into the function
It's bad style, so I'll refactor it.